libs/processors/src/test/kotlin/org/wordpress/android/processor/RemoteConfigProcessorTest.kt

```diff
@@ -119,10 +118,14 @@ class RemoteConfigProcessorTest {
 
     private fun compile(src: List<SourceFile>) = KotlinCompilation().apply {
         sources = src + fakeAppConfig
-        symbolProcessorProviders = listOf(RemoteConfigProcessorProvider())
-        kspWithCompilation = true
         inheritClassPath = true
+        verbose = true
         messageOutputStream = System.out
+        configureKsp(useKsp2 = false) {
+            symbolProcessorProviders += RemoteConfigProcessorProvider()
+            withCompilation = true
+            languageVersion = "1.9"
```

Comment:
Additional reference:

> To use KSP, you must set KotlinCompilation.languageVersion to 1.9.

https://github.com/ZacSweers/kotlin-compile-testing/pull/196#issuecomment-1819608412

Reply:
Awesome, nice find @wzieba ! 🥇

Reply:
Btw, I have no idea why you "must" set `KotlinCompilation.languageVersion to 1.9` for this to work, if you know, please share some more details, since this looks weird to me, and I am not sure that I am missing something basic here... 🤔

Reply:
AFAIU this is because KSP1 was meant to support language features up Kotlin 2.0+ (not included). So if `-language-version` compiler option (this is what `KotlinCompilation.languageVersion` sets) is set to higher, I'm guessing Kotlin Compiler forcefully chooses KSP2, which doesn't work with out processor. This is just a theory though, understanding it fully would diving into source of this logic or asking the author. Or find a way to instruct Kotlin compiler to print out which KSP it uses, but I haven't found a way.

Reply:
https://github.com/ZacSweers/kotlin-compile-testing/discussions/309 🙂

Reply:
Thanks for sharing this theory @wzieba , it makes (some) sense indeed... 👍 + 🤔

Reply:
We got the answer, I think it's aligned with what I described above 👍

Reply:
Yea, 100%, great to have clarity and verification on that! 💯